src/op_analytics/cli/subcommands/pulls/defillama.py

```
@@ -88,60 +84,102 @@ def process_breakdown_stables(data):
+
+    urls = {
+        stablecoin_id: BREAKDOWN_ENDPOINT.format(id=stablecoin_id)
+        for stablecoin_id in stablecoin_ids
+    }
+
+    stablecoin_data = run_concurrently(
+        lambda x: get_data(session, x), urls, max_workers=4
+    )
 
-    stablecoin_data = run_concurrently(lambda x: get_data(session, x), urls, max_workers=4)
+    breakdown_dfs: List[pl.DataFrame] = []
+    metadata_rows: List[Dict[str, Optional[str]]] = []
 
-    breakdown_dfs = []
-    metadata_rows = []
     for data in stablecoin_data.values():
-        breakdown_df, metadata = process_breakdown_stables(data)
-        breakdown_dfs.append(breakdown_df)
-        metadata_rows.append(metadata)
-
-    breakdown_df = pl.concat(breakdown_dfs, how="diagonal_relaxed")
-    metadata_df = pl.DataFrame(metadata_rows, infer_schema_length=len(metadata_rows))
+        breakdown_df, metadata = process_breakdown_stables(data, days=days)
+        if not breakdown_df.is_empty():
+            breakdown_dfs.append(breakdown_df)
+        if metadata:
+            metadata_rows.append(metadata)
+
+    breakdown_df = (
+        pl.concat(breakdown_dfs, how="diagonal_relaxed")
+        if breakdown_dfs
+        else pl.DataFrame()
+    )
+    metadata_df = (
+        pl.DataFrame(metadata_rows, infer_schema_length=len(metadata_rows))
+        if metadata_rows
+        else pl.DataFrame()
+    )
 
-    # Write metadata to BQ
     dt = now_date()
-    overwrite_unpartitioned_table(metadata_df, BQ_DATASET, f"{METADATA_TABLE}_latest")
-    overwrite_partition_static(metadata_df, dt, BQ_DATASET, f"{METADATA_TABLE}_history")
-
-    # Write breakdown to BQ. Only the most recent 7 days are inserted.
-    # Data in BQ older than 7 days is considered immutable.
-    overwrite_partitions_dynamic(
-        df=most_recent_dates(breakdown_df, n_dates=7),
-        dataset=BQ_DATASET,
-        table_name=f"{BREAKDOWN_TABLE}_history",
-    )
+
+    if not metadata_df.is_empty():
```

we should never get empty dataframes should we?  If we do then should we raise an exception instead of silently not writing to BQ?